controlplane/eks/controllers/awsmanagedcontrolplane_controller.go

```diff
@@ -326,3 +351,41 @@ func (r *AWSManagedControlPlaneReconciler) managedClusterToManagedControlPlane(o
 		},
 	}
 }
+
+func (r *AWSManagedControlPlaneReconciler) dependencyCount(ctx context.Context, managedScope *scope.ManagedControlPlaneScope) (int, error) {
+	clusterName := managedScope.Name()
+	namespace := managedScope.Namespace()
+	r.Log.Info("looking for EKS cluster dependencies", "cluster", clusterName, "namespace", namespace)
+
+	listOptions := []client.ListOption{
+		client.InNamespace(namespace),
+		client.MatchingLabels(map[string]string{clusterv1.ClusterLabelName: clusterName}),
+	}
+
+	dependencies := 0
+
+	machines := &infrav1.AWSMachineList{}
+	if err := r.Client.List(ctx, machines, listOptions...); err != nil {
+		return dependencies, fmt.Errorf("failed to list machines for cluster %s/%s: %w", namespace, clusterName, err)
+	}
+	r.Log.V(2).Info("tested for AWSMachine dependencies", "count", len(machines.Items))
+	dependencies += len(machines.Items)
+
+	if feature.Gates.Enabled(feature.MachinePool) {
```

Comment:
I'm wondering if it's worth perhaps just using the API to check for both nodegroups _and_ fargate profiles (which also prevent deletion)? Or are dependencies created outside of CAPI out of scope here?

Reply:
I did think about that but my thinking is that the CAPA CRD types are the dependency which in turn own the AWS resources and I didn't want the control plane to be aware of the implementation details of each of the other controllers.

Reply:
Yeah that's totally valid 👍